src/gsi/radarZ_module.f90

```diff
@@ -0,0 +1,2887 @@
+!
+! PURPOSE:
+!
+! This subroutine sets intercept parameters for rain/snow/hail and
+! densities for snow/hail based on values in history dump.
+!
+!-----------------------------------------------------------------------
+!
+! AUTHOR:  Youngsun Jung, Spring 2010
+!
+!-----------------------------------------------------------------------
+! Include global variables only for dual-pol calculations
+!-----------------------------------------------------------------------
+
+  USE radarz_cst, only: mphyopt,n0rain,n0snow,n0hail,n0grpl
+  USE radarz_cst, only: rhosnow,rhohail,rhogrpl
+  USE radarz_cst, only: alpharain,alphasnow,alphagrpl,alphahail
+
+!-----------------------------------------------------------------------
+! Force explicit declarations.
+!-----------------------------------------------------------------------
+
+  IMPLICIT NONE
+
+!-----------------------------------------------------------------------
+! Include files.
+!-----------------------------------------------------------------------
+
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+!
+! Beginning of executable code...
+!
+!@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@@
+
+  CALL model_dsd(n0rain,n0snow,n0hail,n0grpl,rhosnow,rhohail,rhogrpl,  &
+       alpharain,alphasnow,alphagrpl,alphahail)
+
+  IF (rhos <= 0.0_r_kind) THEN
+    rhos = 100._r_kind
+  END IF
+
+  IF (rhoh <= 0.0_r_kind) THEN
+    rhoh = 913._r_kind
+  END IF
+
+  IF (rhog <= 0.0_r_kind) THEN
+
+    SELECT CASE (mphyopt)
+      CASE(1:12,108:110)
+        rhog = 400._r_kind
+      CASE(106,116)
+        rhog = 500._r_kind
+    END SELECT
+
+  END IF
+
+  IF (N0r <= 0.0_r_kind) THEN
+    N0r = 8.0E+06_r_kind
+  END IF
+
+  IF (N0s <= 0.0_r_kind) THEN
+    N0s = 3.0E+06_r_kind
+    SELECT CASE (mphyopt)
+      CASE(1:12,106,108:110,116)
+        N0s2 = 0.0_r_kind
+    END SELECT
+  END IF
+
+  IF (N0h <= 0.0_r_kind) THEN
+    N0h = 4.0E+04_r_kind
+  END IF
+
+  IF (N0g <= 0.0_r_kind) THEN
+    SELECT CASE (mphyopt)
+      CASE(1:12,108:110)
+        N0g = 4.0E+05_r_kind
+      CASE(106,116)
+        N0g = 4.0E+06_r_kind
+    END SELECT
+  END IF
+
+   N0ms = N0s
+   N0ms2 = N0s2
+   N0mh = N0h
+   N0mg = N0g
+
+  IF (alphar <= 0.0_r_kind) THEN
+    SELECT CASE (mphyopt)
+      CASE(1:12,106,108:110)
+        alphar = 0.0_r_kind
+    END SELECT
+   END IF
+
+   IF (alphas <= 0.0_r_kind) THEN
+       alphas = 0.0_r_kind
+   END IF
+
+   SELECT CASE (mphyopt)
+     CASE(1:12,106,109,110,116)
+       alphas2 = 0.0_r_kind
+     CASE(108)
+       alphas2 = 0.6357_r_kind
+   END SELECT
+
+   IF (alphah <= 0.0_r_kind) THEN
+     alphah = 0.0_r_kind
+   END IF
+
+   IF (alphag <= 0.0_r_kind) THEN
+      alphag = 0.0_r_kind
+   END IF
+
+   lamdar = 0.0_r_kind
+   lamdas = 0.0_r_kind
+   lamdas2 = 0.0_r_kind
+   lamdams = 0.0_r_kind
+   lamdams2 = 0.0_r_kind
+   lamdag = 0.0_r_kind
+   lamdamg = 0.0_r_kind
+   lamdah = 0.0_r_kind
+   lamdamh = 0.0_r_kind
```

Comment:
Please align END IF with IF on line 2729.

Reply:
We aligned the code as you suggested.

https://github.com/NOAA-EMC/GSI/pull/142/files#diff-fa0ecb3836c32fcbab1f64a2be08a2292b9c3414d702d475609b32f8205ffcf1R2724-R2763